### DISCUSSION TITLE
A suggestion to make the code more idiomatic

### DISCUSSION THREAD

**Discussion by Daniel-Silveira** [kaiosilveira/extract-function-refactoring#4] [Q&A] [answered]
Hi @kaiosilveira !

I was reading this refactoring and noticed that this logic to calculate the outstanding amount:

https://github.com/kaiosilveira/extract-function-refactoring/blob/2935484234efdee68c954a341bfdda7a4d03fc5f/index.js#L28

could be potentially refactored forward to be more JS idiomatic, probably using a `.map().reduce()` operation, which is more common in the JS world. What do you think about it?

**Comment by kaiosilveira** [comment, marked as answer]
Hey Dan!

Thanks for sharing your thoughts!

Yeah, I completely agree with you. For now, I'm basically implementing the code exactly as it is in the book so I don't get lost halfway through the refactoring and readers that also read the book will not notice any differences. Nevertheless, I could create a new section at the end of the refactoring docs, something like "additional refactorings" or "departing thoughts", etc.

Thank you for the suggestion. 👊🏽
